src/Client.php

```diff
@@ -38,9 +44,39 @@ public function getIndexer(): Indexer
 
     public function getIndex($name): Index
     {
+        $name = $this->getPrefixedIndex($name);
+
         return new Index($this, $name);
     }
 
+    public function getPrefixedIndex($pureName): string
+    {
+        $prefix = $this->getConfigValue(self::CONFIG_INDEX_PREFIX);
+
+        if ($prefix) {
+            return sprintf('%s_%s', $prefix, $pureName);
+        } else {
+            return $pureName;
+        }
+    }
+
+    public function getPureIndexName($indexName): string
+    {
+        $prefix = $this->getConfigValue(Client::CONFIG_INDEX_PREFIX);
+
+        if ($prefix) {
+            $pattern = sprintf('/%s_(.+)_\d{4}-\d{2}-\d{2}-\d+/i', preg_quote($prefix, '/'));
```

Comment:
I would not hardcode the `_` here, and let it to the responsibility of the developer

Reply:
It make it easier to force it, for regexp, and to make sure index name is always readable.

Reply:
I don't like it, but OK :+1:

Reply:
Just one question : we could use https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-meta-field.html to store more information and keep index name clean

Reply:
:+1: for `_meta`, but I think it must left some bytes to avoid collisions anyway

Reply:
This is a nice idea, but it would mean querying Elasticsearch for the Index (HTTP GET query). As we need the Pure Index name for each doc of a resultSet (buildModelFromIndexAndData), this is not acceptable.